File: android/src/main/kotlin/com/emarsys/emarsys_sdk/flutter/InlineInAppViewFactory.kt
package com.emarsys.emarsys_sdk.flutter

import android.content.Context
import com.emarsys.emarsys_sdk.ui.InlineInAppView
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.StandardMessageCodec
import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory

class InlineInAppViewFactory(
    private val messenger: BinaryMessenger,
    private val applicationContext: Context
) :
    PlatformViewFactory(StandardMessageCodec.INSTANCE) {

    override fun create(context: Context?, viewId: Int, args: Any?): PlatformView {
        val contextForInlineInAppView = context ?: applicationContext
        val creationParams = args as Map<String?, Any?>?
        return InlineInAppView(contextForInlineInAppView, viewId, creationParams,messenger)
    }
}

File: ios/Classes/Flutter/InlineInAppViewFactory.swift
//
//  Created by Emarsys
//

import Foundation
import Flutter
import UIKit
import EmarsysSDK

class InlineInAppViewFactory: NSObject, FlutterPlatformViewFactory {
    private var messenger: FlutterBinaryMessenger

    init(messenger: FlutterBinaryMessenger) {
        self.messenger = messenger
        super.init()
    }
    
    public func createArgsCodec() -> FlutterMessageCodec & NSObjectProtocol {
          return FlutterStandardMessageCodec.sharedInstance()
    }
    
    func create(
        withFrame frame: CGRect,
        viewIdentifier viewId: Int64,
        arguments args: Any?
    ) -> FlutterPlatformView {
        return InlineInAppView(
            frame: frame,
            viewIdentifier: viewId,
            arguments: args as? Dictionary<String, String> ?? Dictionary(),
            binaryMessenger: messenger)
    }
}


